cmd/security-bootstrapper/entrypoint-scripts/redis_wait_install.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+#  ----------------------------------------------------------------------------------
+#  Copyright (c) 2021 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#  SPDX-License-Identifier: Apache-2.0
+#  ----------------------------------------------------------------------------------
+
+# This is customized entrypoint script for Redis.
+# In particular, it waits for the TokensReady Port being ready to roll
+
+set -e
+
+# env settings
+. /edgex-init/.env-redis
+
+echo "Script for waiting security bootstrapping installation"
+
+# gating on the TokensReadyPort
+echo "$(date) Executing dockerize on Redis with waiting on TokensReadyPort tcp://${VAULTWORKER_HOST}:$WAIT_TCP_PORT"
+/edgex-init/dockerize -wait tcp://"${VAULTWORKER_HOST}":"$WAIT_TCP_PORT" -timeout 60s
+
+# the bootstrap-redis needs the connection from Redis db to set it up.
+# Hence, here bootstrap-redis runs in background and then starts the Redis db.
+echo "$(date) ${VAULTWORKER_HOST} tokens ready, bootstrapping redis..."
+/edgex-init/bootstrap-redis/security-bootstrap-redis --confdir=/edgex-init/bootstrap-redis/res &
+
+# give some time for bootstrap-redis to start up
+sleep 1
+
+echo "$(date) Starting edgex-redis..."
+exec /usr/local/bin/docker-entrypoint.sh redis-server &
+
+# signal the redis is ready
+# in a forever loop so that it keeps listening all the times
+while true; do
+  /edgex-init/security-bootstrapper --confdir=/edgex-init/res listenTcp \
```

Comment:
Not sure why this is needed.  Why not have ready_to_run just wait on the redis port directly?

Reply:
Good question: because there is a time gap between redis port opens and the `redis-bootstrapper` has done the bootstrapping.  Hence, need another port to signal that.  With that being said, I missed that waiting implementation.
And I'll put that in. Thanks!

Reply:
Bryon: Added the wait implementation.